lib/suse/connect/hwinfo/arm64.rb

```diff
@@ -0,0 +1,62 @@
+# Collect hardware information for aarch64 systems
+class SUSE::Connect::HwInfo::ARM64 < SUSE::Connect::HwInfo::Base
+  class << self
+    def hwinfo
+      {
+        hostname: hostname,
+        cpus: cpus,
+        sockets: sockets,
+        hypervisor: hypervisor,
+        arch: arch,
+        uuid: uuid
+      }
+    end
+
+    def cpus
+      output['CPU(s)'].to_i
+    end
+
+    def sockets
+      output['Socket(s)'].to_i
+    end
+
+    def hypervisor
+      vendor = execute('systemd-detect-virt -v', false, [0, 1])
+      vendor == 'none' ? nil : vendor
+    end
+
+    def uuid
+      if respond_to?("#{arch}_uuid", true)
```

Comment:
> There is no need to check if this method is defined, it is defined 10 lines below.

Right. I took the `x86.rb` as a template :wink: But it makes no sense there either.
I guess, we can remove it their as well?
It would also make no sense in the `base.rb` as without a known architecture it does not even try to find the uuid.